Use a shared_ptr for the dynamic reconfigure pointer, and create it w…

lucasw opened the pull request:
…ith the private node handle so that the parameters for the dynamic reconfigure server are in the private namespace and two image publishers can coexist in the same manager #357 

This may break parameters that were set in the nodelet manager  namespace as a workaround in some launch files out there.

lucasw committed:
Use a shared_ptr for the dynamic reconfigure pointer, and create it with the private node handle so that the parameters for the dynamic reconfigure server are in the private namespace and two image publishers can coexist in the same manager #357

JWhitleyWork commented:
@lucasw - Thanks for the PR! I know this is a bit old but, would you mind removing your custom launch file and rebasing this on `melodic`? Also, I'm OK with the use of std::shared_ptr<>. We should be upgrading to it across the code-base for `melodic` anyway.

lucasw commented:
Wouldn't you rather take this as a boost shared_ptr, then have a separate PR that switches out all the boost shared_ptrs for all of image_pipeline?  (created an issue #407)  It seems odd to just have one std shared_ptr and then 40 boost shared_ptrs.

JWhitleyWork commented:
@lucasw - That's a good point. Thanks for the update. Would you mind rebasing this on `melodic` to grab the new CI? After, I'll have one other person review and then we'll merge.

lucasw commented:
I've removed the launch file, rebased to melodic, and squashed.  I left a backup of the launch file for myself in https://github.com/lucasw/image_pipeline/tree/indigo_image_pub_dr_private_namespace .
